compiler/src/codegen/transl_anf.re

```diff
@@ -1228,44 +1235,78 @@ let lift_imports = (env, imports) => {
   (imports, setups, env);
 };
 
-let transl_signature = (functions, signature) => {
+let transl_signature = (functions, imports, signature) => {
   open Types;
 
+  let function_exports = ref([]);
+
   // At this point in compilation, we know which functions can be called
   // directly/indirectly at the wasm level. We add this information to the
   // module signature.
   let func_map = Ident_tbl.create(30);
   List.iter(
-    func => Ident_tbl.add(func_map, (func: mash_function).id, func),
+    (func: mash_function) =>
+      switch (func.name) {
+      | Some(name) =>
+        Ident_tbl.add(func_map, func.id, Ident.unique_name(func.id))
+      | None => ()
+      },
     functions,
   );
+  List.iter(
+    imp =>
+      switch (imp.imp_shape) {
+      | FunctionShape(_) =>
+        let internal_name =
+          switch (imp.imp_desc) {
+          | GrainValue(mod_, name) => grain_import_name(mod_, name)
+          | WasmFunction(mod_, name) => Ident.unique_name(imp.imp_use_id)
+          | _ => failwith("Wasm or js value had FunctionShape")
```

Comment:
Is this something users can run into? Should we have a more descriptive error?

Reply:
Nope! It's not possible, so I'll add an "Impossible" marker to it. We can maybe make the types a little better here in the future to avoid having a case like this (combining imp_desc and imp_shape).